packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -182,8 +182,11 @@ contract AssignmentHook is EssentialContract, IHook {
         pure
         returns (uint256)
     {
-        for (uint256 i; i < tierFees.length; ++i) {
+        for (uint256 i; i < tierFees.length;) {
             if (tierFees[i].tier == tierId) return tierFees[i].fee;
+            unchecked {
+                ++i;
+            }
```

Comment:
No need to have this optimizaton, bc. fro 0.8.22:

<img width="740" alt="kép" src="https://github.com/taikoxyz/taiko-mono/assets/51912515/e0dc8f35-1d99-441e-82e0-7a3c01653dfb">

Unchecked loop increemnt optimizer:
<img width="780" alt="kép" src="https://github.com/taikoxyz/taiko-mono/assets/51912515/6b98def4-cfdd-4e9a-887b-20b49ebaa343">
https://docs.soliditylang.org/en/latest/internals/optimizer.html#unchecked-loop-optimizer

Reply:
Good to know.